rfcs/debuggability_metrics.md

```diff
@@ -0,0 +1,126 @@
+   3. Add breakpoints. This is a less common step.
+   4. Add new output. This is a less common step.
+   5. Run the appropriate executable(s).
+   6. Run the appropriate test(s).
+4. Make changes to actually fix the bug.
+   * If no test existed to catch the bug, also add an appropriate test(s) to prevent the bug from regressing.
+
+### Measuring Debugging Time
+
+Each of the steps above can be measured in time, although the value of those data points will vary wildly. Many of them are too dependent on the bug, can be done in parallel, or are out of scope for debugability. In order to ensure consistent measuring, we will perform a UX study where research subjects are monitored by a researcher. The researcher will measure the time each step takes with a stopwatch, record the results, and collect other feedback.
+
+Local reproduction is a borderline case - things like build time are out of scope, but making sure that the bug can be reproduced locally reliably are in scope. Further, there's a ton of variables involved that will make any metrics we do measure exceptionally noisy. Did they have a partial sync already? Did they have a build already? Do they prefer to debug via logs or debug via the debugger? All of these will cause the time to vary wildly while not actually indicating the ease of the debugging as defined as in-scope for this RFC. Given that local reproduction is more of a binary question (Can we reproduce the bug locally?) given the scope of this RFC, we will not be including it as a suggested timed metric.
+
+### Debugging Time Metrics
+
+Thus, this leaves us with 3 areas to measure for time:
+
+1. Error reporting time: This step would have three parts. Currently, Jenkins only outputs an error e-mail when all builds in a run are finished, but changes could be made to report an issue as soon as a build in a run has failed.
+   1. Time when failure is first signaled inside Jenkins job
+   2. Time when Jenkins sends an e-mail reporting the failure to the user (or a bot posts a failure message)
+   3. Time when user takes action on the notification and starts working on their machine (modifying code, or starting a build)
+2. Log-finding time: This step would have two parts. We expect this to be a small portion of the total time consumed, but it is still valuable to reduce it where possible.
+   1. Time it takes from notification to finding the correct log
+   2. Time it takes (after finding the correct log) to parse enough information in the log to take action locally
+3. Artifact-finding time: We expect this to be a small portion of the total time consumed, but it is still valuable to reduce it where possible.
+
+The first is a binary option - we can either send the e-mail at the end of the build, or upon the first failure. But measuring that difference could allow us to decide if the earlier messaging time is worthwhile. The second two options are going to have to be user-reported metrics.
+
+Metric | Description | Measured By
```

Comment:
Another metric worth considering is when logs are found, are they actually useful in debugging the issue.